src/types/token.ts

```diff
@@ -0,0 +1,30 @@
+// External Libraries
+import * as _ from "lodash";
+
+// Utils
+import { BigNumber } from "../../utils/bignumber";
+import { TOKEN_REGISTRY_TRACKED_TOKENS } from "../../utils/constants";
+
+export class Token {
+    public readonly address: string;
+    public readonly decimals: BigNumber;
+    public readonly name: string;
+    public readonly symbol: string;
+
+    constructor(symbol: string) {
+        this.symbol = symbol;
+
+        const token = _.find(
+            TOKEN_REGISTRY_TRACKED_TOKENS,
+            (tokenObject) => tokenObject.symbol === symbol,
```

Comment:
The naming here is confusing, around `token` and `tokenObject` given that you're already in a class called `Token`. I would refer to the objects mapped here as something like:

const `attributes = _.find(TOKEN_REGISTRY_TRACKED_TOKENS, (tokenInfo) => tokenInfo.symbol...`